Documentation/cassandra-operator-upgrade.md

```diff
@@ -0,0 +1,74 @@
+1. The main controller that runs as a standalone application, watches the Cassandra Cluster CRD and makes administrative decisions. We'll call this component `Controller` for the rest of this document.
+2. A sidecar that runs alongside each member of a Cassandra Cluster. We will call this component `Sidecar`.
+
+Both components should be updated. This is a very manual process at the moment, but it should be automated soon in the future, once the Cassandra Operator reaches the beta stage.
+
+## Considerations
+With this upgrade guide, there are a few notes to consider:
+* **WARNING:** Upgrading a Rook cluster is not without risk. There may be unexpected issues or
+  obstacles that damage the integrity and health of your storage cluster, including data loss. Only
+  proceed with this guide if you are comfortable with that. It is recommended that you backup your data before proceeding.
+* **WARNING:** The current process to upgrade REQUIRES the cluster to be unavailable for the time of the upgrade.
+
+## Prerequisites
+
+* Before starting the procedure, ensure that your Cassandra Clusters are in a healthy state. You can check a Cassandra Cluster's health by using `kubectl describe $NAME -n $NAMESPACE` and ensuring that for each rack in the Status, `readyMembers` equals `members`.
+* If you are upgrading from v0.9.1 to a later version, the mount point of the PVC for each member has changed because it was wrong. 
+
+## Procedure
+
+1. Because each version of the `Controller` is designed to work with the same version of the `Sidecar`, they must be upgraded together. In order to avoid mixing versions between the `Controller` and `Sidecar`, we first delete every Cassandra Cluster CRD in our Kubernetes cluster, after first backing up their manifests. This will not delete your data because the PVCs will be retained even if the Cassandra Cluster object is deleted. Example:
+```bash
+# Assumes cluster rook-cassandra in namespace rook-cassandra
+NAME=rook-cassandra
+NAMESPACE=rook-cassandra
+
+kubectl get $NAME -n $NAMESPACE -o yaml > $NAME.yaml
+kubectl delete $NAME -n $NAMESPACE
+```
+
+2. After that, we upgrade the version of the `Controller`. To achieve that, we patch the StatefulSet running the `Controller`:
+```bash
+# Assumes controller is running in StatefulSet rook-cassandra-operator
+# in namespace rook-cassandra-system
+
+kubectl -n rook-cassandra-system set image sts/rook-cassandra-operator rook-cassandra-operator=rook/cassandra:v0.9.x
+```
+
+After patching, ensure that the operator pods are running successfully:
+```bash
+kubectl get pods -n rook-cassandra-system
+```
+
+3. Recreate the manifests previously deleted:
+```bash
+kubectl apply -f $NAME.yaml
+```
+The controller will pick up the newly created Cassandra Clusters and recreate them with the correct version of the sidecar.
+
+
+## Before Upgrading from v0.9.1
```

Comment:
It would be good to add a note at the top of the document that this section exists and must be walked through if one wants to upgrade to v0.9.1

I think the anchor for this section would be `#before-upgrading-from-v091`.

Reply:
Sounds good, will do!